src/lib/db/context-field-store.ts

```diff
@@ -19,8 +22,30 @@ const mapRow = row => ({
     createdAt: row.created_at,
 });
 
+export interface ICreateContextField {
+    name: string;
+    description: string;
+    stickiness: boolean;
+    sort_order: number;
+    legal_values?: string[];
+    updated_at: Date;
+}
+
+export interface IContextField {
+    name: string;
+    description: string;
+    stickiness: boolean;
+    sortOrder: number;
+    legalValues?: string[];
+    createdAt: Date;
+}
+
 class ContextFieldStore {
-    constructor(db, customContextFields, getLogger) {
+    private db: Knex;
+
+    private logger: Logger;
+
+    constructor(db: Knex, customContextFields, getLogger: LogProvider) {
```

Comment:
What should be the type of `customContextFields` here?

Reply:
we should just remove these in v4. not needed when we have a proper API.